src/components/Apps/AppsList/AppsList.tsx

```diff
@@ -1,17 +1,131 @@
+// supports both. TODO: Check if there are any catalogs left with the old label
+// if not, simplify this code.
+const isInternal = (catalog: IAppCatalog): boolean => {
+  const labels = catalog.metadata.labels;
+
+  if (labels) {
+    const catalogType = labels['application.giantswarm.io/catalog-type'];
+    const catalogVisibility =
+      labels['application.giantswarm.io/catalog-visibility'];
+
+    return catalogType === 'internal' || catalogVisibility === 'internal';
+  }
+
+  return false;
+};
+
+// Admins can see all catalogs.
+// Non admins can only see non internal catalogs.
+const filterFunc = (isAdmin: boolean) => {
+  return ([_, catalog]: [string, IAppCatalog]) => {
+    if (isAdmin) {
+      return true;
+    }
+
+    return !isInternal(catalog);
+  };
+};
+
+// Group internal catalogs together, but sort the groups alphabetically.
+// This makes internal catalogs appear at the bottom, and normal catalogs
+// appear at the top. And within those groups the catalogs are alphabetically
+// sorted.
+const sortFunc = (
+  [, a]: [string, IAppCatalog],
+  [, b]: [string, IAppCatalog]
+) => {
+  const aIsInternal = isInternal(a);
+  const aTitle = a.spec.title;
+
+  const bIsInternal = isInternal(b);
+  const bTitle = b.spec.title;
+
+  if (aIsInternal && !bIsInternal) {
+    return 1;
+  } else if (!aIsInternal && bIsInternal) {
+    return -1;
+  }
+  if (aTitle < bTitle) {
+    return -1;
+  } else if (aTitle > bTitle) {
+    return 1;
+  }
+
+  return 0;
+};
+
+function catalogsToFacets(
+  catalogs: IAppCatalogsState,
+  isAdmin: boolean
+): IFacetOption[] {
+  return Object.entries(catalogs.items)
+    .filter(filterFunc(isAdmin))
+    .sort(sortFunc)
+    .map(([key, catalog]) => {
+      return {
+        value: key,
+        checked: catalogs.ui.selectedCatalogs[key],
+        label: (
+          <CatalogLabel
+            catalogName={catalog.spec.title}
+            iconUrl={catalog.spec.logoURL}
+          />
+        ),
+      };
+    });
+}
+
+const AppsList: React.FC<IAppsListProps> = (props) => {
+  const dispatch = useDispatch();
 
-const AppsList: React.FC = () => {
   return (
     <AppsListPage
       matchCount={0}
       onChangeFacets={(value, checked) => {
-        console.log(value, checked);
+        if (checked) {
+          dispatch(enableCatalog(value));
+        } else {
+          dispatch(disableCatalog(value));
+        }
       }}
       apps={[]}
-      facetOptions={[]}
+      facetOptions={catalogsToFacets(props.catalogs, props.isAdmin)}
     />
   );
 };
 
-export default AppsList;
+function mapStateToProps(state: IState) {
+  return {
+    catalogs: state.entities.catalogs,
+    isAdmin: getUserIsAdmin(state),
+  };
+}
+
+AppsList.propTypes = {
+  catalogs: (PropTypes.object as PropTypes.Requireable<IAppCatalogsState>)
+    .isRequired,
+  isAdmin: PropTypes.bool.isRequired,
+};
+
+export default connect(mapStateToProps)(AppsList);
```

Comment:
This should not change functionality, but in regards to consistency to other files: Why not use `useSelector` here, since we're using a functional component?

https://github.com/giantswarm/happa/blob/273cb78609fb34a4286c7a171d5d0bb2f1294713/src/components/Cluster/ClusterDetail/Ingress/InstallIngressButton.tsx#L62

Reply:
Not used to it yet.. kinda went with what I remembered and saw. But I think it makes sense to use `useSelector` I'll apply that before merging.